src/engine/input/key_mapping/serialisation/NamedInputGroup.cs

```diff
@@ -0,0 +1,20 @@
+using System.Collections.Generic;
+using System.Linq;
+
+public class NamedInputGroup
+{
+    public string GroupName { get; set; }
+    public IReadOnlyList<string> EnvironmentId { get; set; }
+    public IReadOnlyList<NamedInputAction> Actions { get; set; }
+
+    public InputGroupItem ToGodotObject(InputDataList data)
```

Comment:
So this builds an object out of this? This way of building objects isn't used as such for the other stuff.
I think it would be better if the root object of the options change hierarchy created its children, instead of doing stuff like this. Because if someone calls ToGodotObject on this and doesn't do anything with the result, that will leak the godot Node.